src/util/misc/matrix.ts

```diff
@@ -107,22 +106,126 @@ export const qrDecompose = (a: TMat2D): TQrDecomposeOut => {
 };
 
 /**
- * Returns a transform matrix starting from an object of the same kind of
- * the one returned from qrDecompose, useful also if you want to calculate some
- * transformations from an object that is not enlived yet
- * @param  {Object} options
- * @param  {Number} [options.angle] angle in degrees
- * @return {TMat2D} transform matrix
+ * Generate a translation matrix
+ *
+ * A translation matrix in the form of
+ * [ 1 0 x ]
+ * [ 0 1 y ]
+ * [ 0 0 1 ]
+ *
+ * See @link https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/transform#translate for more details
+ *
+ * @param {number} x translation on X axis
+ * @param {number} [y] translation on Y axis
+ * @returns {TMat2D} matrix
  */
-export const calcRotateMatrix = ({ angle }: TRotateMatrixArgs): TMat2D => {
-  if (!angle) {
-    return iMatrix;
-  }
-  const theta = degreesToRadians(angle),
-    cosin = cos(theta),
-    sinus = sin(theta);
-  return [cosin, sinus, -sinus, cosin, 0, 0];
-};
+export const calcTranslateMatrix = (x: number, y = 0): TMat2D => [
+  1,
+  0,
+  0,
+  1,
+  x,
+  y,
+];
+
+/**
+ * Generate a rotation matrix around around a point (x,y), defaulting to (0,0)
+ *
+ * A matrix in the form of
+ * [cos(a) -sin(a) -x*cos(a)+y*sin(a)+x]
+ * [sin(a)  cos(a) -x*sin(a)-y*cos(a)+y]
+ * [0       0      1                 ]
+ *
+ *
+ * @param {TDegree} angle rotation in degrees
+ * @param {number} [x] translation on X axis for the pivot point
+ * @param {number} [y] translation on Y axis for the pivot point
+ * @returns {TMat2D} matrix
+ */
+export function calcRotateMatrix(
+  { angle = 0 }: TRotateMatrixArgs = {},
+  x = 0,
+  y = 0
+): TMat2D {
+  const angleRadiant = degreesToRadians(angle),
+    cosValue = cos(angleRadiant),
+    sinValue = sin(angleRadiant);
+  return [
+    cosValue,
+    sinValue,
+    -sinValue,
+    cosValue,
+    x - (cosValue * x - sinValue * y),
+    y - (sinValue * x + cosValue * y),
+  ];
+}
```

Comment:
The parser calcRotateMatrix and the old calcRotateMatrix are the only one i don't want to mix.
I m not sure the pivot point in our geometry is identical to the one in the svg context.

Reply:
I will add a test comparing this and rotatePoint

Reply:
take a look at 5d3642811 with critical eyes

Reply:
i m not worried about that use case.
Those matrices were the building block of calcTransformMatrix.
I would like to have the calcRotateMatrix used for calcTransformMatrix, different from the function that has to only serve the SVG parsing abilities.

Is ok if the svg parser call the old `calcRotateMatrix` and then adds the missin 2 values and we name that parseSVGRotateTransformAttribute.

I don't know of svg notations that may need different operation to the one that we have and that are identical to the one we were already using, so scale/skew/translate do not bother me.

Reply:
I don't really understand
I think I proved it is identical to how we use origin and since there aren't other use cases seems ok.
But I understand the mistrust.
I do think it is useful outside the parsing context.

Reply:
i just want the pivot point, that was part of the svg parsing and nothing else, to stay part of the svg parsing.
I don't want to mix them up because we save a bunch of code and because if we pass 0,0 at the pivot point then the function is functionally he same.
Let's leave the previous calcRotateMatrix and the svg one divided.
We may want to change the svg parsing code independently from the canvas code.

Reply:
I still don't get your point. As a standalone util it is great, isn't it?
Are you worried it does more calculations than before? Is that the point?

> We may want to change the svg parsing code independently from the canvas code.

Then we change it accordingly.

I have encountered a number of times the need of this util and had to use `rotatePoint` to figure out the translation. That is why I want it exposed.

Reply:
Well the 4 addition and 4 multiplication are also unwelcome an extra point to have divided in 2 stages that is all i m asking.

Reply:
can I just have an early return instead if the origin is 0,0?